src/lib/backend_fapi.c

```diff
@@ -0,0 +1,490 @@
+        if (rc) {
+            LOGE("Getting the TPM data blobs failed.");
+            goto error;
+        }
+
+        pub = twistbin_new(tpm2bPublic, tpm2bPublicSize);
+        Fapi_Free(tpm2bPublic);
+        priv = twistbin_new(tpm2bPrivate, tpm2bPrivateSize);
+        Fapi_Free(tpm2bPrivate);
+        if (!pub || !priv) {
+            LOGE("Out of memory");
+            goto error;
+        }
+
+        t->sealobject.sopub = pub;
+        t->sealobject.sopriv = priv;
+
+        rc = Fapi_GetAppData(t->fapi.ctx, path, &appdata, &appdata_len);
+        if (rc) {
+            LOGE("Getting FAPI seal appdata failed.");
+            goto error;
+        }
+
+        t->sealobject.soauthsalt = twistbin_new(appdata, strlen((char *)appdata));
+        if (t->sealobject.soauthsalt == NULL) {
+            LOGE("OOM");
+            goto error;
+        }
+
+        yaml = appdata + strlen((char *)appdata) + 1;
+
+        while ((size_t)(yaml - appdata) < appdata_len) {
+            LOGV("Current yaml at offset %zi / %zi is: %s",
+                 yaml - appdata, appdata_len, yaml);
+
+            tobject *tobj = tobject_new();
+            if (!tobj) {
+                LOGE("oom");
+                Fapi_Free(appdata);
+                goto error;
+            }
+
+            tobj->id = objectids;
+            objectids += 1;
+
+            if (!parse_attributes_from_string(yaml, strlen((char*)yaml), &tobj->attrs)) {
+                LOGE("Could not parse FAPI attrs, got: \"%s\"", yaml);
+                free(tobj);
+                Fapi_Free(appdata);
+                goto error;
+            }
+
+            rv = object_init_from_attrs(tobj);
+            if (rv != CKR_OK) {
+                LOGE("Object initialization failed");
+                free(tobj);
+                Fapi_Free(appdata);
+                goto error;
+            }
+
+            rv = token_add_tobject_last(tok, tobj);
+            if (rv != CKR_OK) {
+                LOGE("Failed to initialize tobject from FAPI");
+                free(tobj);
+                Fapi_Free(appdata);
+                goto error;
+            }
+
+            yaml += strlen((char *)yaml) + 1;
+            LOGV("\nCurrent next is: %zi / %zi", yaml - appdata, appdata_len);
+        }
+        Fapi_Free(appdata);
+
+        rv = tpm_get_existing_primary(t->tctx, &t->pobject.handle, &blob);
+        if (rv != CKR_OK) {
+            return rv;
+        }
+        twist_free(blob);
+
+        t->config.is_initialized = true;
+
+        /* Initialize the User PIN area. */
+        /*********************************/
+        path = tss_path_from_id(t->id, "usr");
+        if (!path) {
+            LOGE("OOM");
+            goto error;
+        }
+        rc = Fapi_GetTpmBlobs(t->fapi.ctx, path, &tpm2bPublic, &tpm2bPublicSize,
+                              &tpm2bPrivate, &tpm2bPrivateSize, NULL /* policy */);
+        if (rc == 0x00060020) {
```

Comment:
That's a very specific RC, is their any RC defines you can OR in to make it more clear?